easybuild/framework/easyblock.py

```diff
@@ -1738,8 +1738,10 @@ def _skip_step(self, step, skippable):
             self.log.info("Skipping %s step (skip: %s, skipsteps: %s)", step, self.skip, self.cfg['skipsteps'])
             skip = True
 
-        # skip step when only generating module file; still run sanity check without use of force
-        elif module_only and not step in ['sanitycheck', 'module']:
+        # skip step when only generating module file
+        # * still run sanity check without use of force
+        # * always run ready & prepare step to set up toolchain + deps
+        elif module_only and not step in ['ready', 'prepare', 'sanitycheck', 'module']:
```

Comment:
need modifications in the help message of module only. so move the list of steps to constants and genereate the help from it